src/Pages/Repositories/PopularRepositoriesTable/PopularRepositoriesTable.tsx

```diff
@@ -399,43 +430,69 @@ const PopularRepositoriesTable = () => {
-                              className={className}
-                              splitButtonItems={[
-                                <DropdownToggleAction
-                                  key='action'
-                                  data-ouia-component-id='add_checked_repos-with-snap'
-                                  onClick={() => addSelected(true)}
-                                  className={className}
-                                >
-                                  {defaultText}
-                                </DropdownToggleAction>,
-                              ]}
-                              toggleVariant='primary'
-                              splitButtonVariant='action'
-                              onToggle={onDropdownToggle}
-                              isDisabled={isDisabled}
-                            />
-                          }
-                          isOpen={isActionOpen}
-                        >
-                          <DropdownItem
-                            key='action'
-                            component='button'
-                            onClick={() => addSelected(false)}
-                            ouiaId='add-selected-repos-no-snap'
+                      // temporarily disable snapshotting by default for popular repos
+                      if (isProd()) {
+                        return (
+                          <Dropdown
+                            onSelect={onDropdownSelect}
+                            className={classes.addRepositoriesButton}
+                            ouiaId='add-selected-toggle-dropdown'
+                            toggle={
+                              <DropdownToggle
+                                ouiaId='add-selected-dropdown-toggle-no-snap'
+                                className={className}
+                                splitButtonItems={[
+                                  <DropdownToggleAction
+                                    key='action'
+                                    data-ouia-component-id='add_checked_repos-without-snap'
+                                    onClick={() => addSelected(false)}
+                                    className={className}
+                                  >
+                                    {defaultText}
+                                  </DropdownToggleAction>,
+                                ]}
+                                toggleVariant='primary'
+                                splitButtonVariant='action'
+                                onToggle={onDropdownToggle}
+                                isDisabled={isDisabled}
+                              />
+                            }
+                            isOpen={isActionOpen}
                           >
-                            {withoutSnapshotText}
-                          </DropdownItem>
-                        </Dropdown>
-                      );
+                            {dropdownItems(defaultText)}
```

Comment:
Curious. Why have a text variable at all if above you are conditionally rendering your dropdown items?
You could easily simplify this a bit, also, instead of using the "isProd()" directly, you have a "isInProd" variable here that is stable. Although this likely should cause any issues, the recheck of the "isProd()" function could cause a few unneeded renders.

Reply:
understood about using the isInProd directly, that makes sense! i'll make that change

regarding the text variable, if i just render the text `Add ${checkedRepositoriesToAdd.size} repositories with / without snapshotting` instead of passing in the text to the dropdownItems, the `checkedRepositoriesToAdd.size` variable always seems to be set to 0. i agree, i don't think i should need to pass in a variable here

Reply:
oh i think i just need to add that value to the dep array :)